src/vectordbs/pg_embedding.rs

```
@@ -84,29 +86,37 @@ impl VectorDb for PgEmbedding {
                 exec_res.rows_affected()
             )));
         };
-        let query = format!(
-            r#"
-                CREATE INDEX ON {INDEX_TABLE_PREFIX}{index_name} USING hnsw(embedding {distance_extension}) WITH (dims={vector_dim}, m={}, efconstruction={}, efsearch={});
-            "#,
-            self.config.m, self.config.efconstruction, self.config.efsearch
-        );
-        let exec_res: ExecResult = self
-            .db_conn
-            .execute(Statement::from_string(DbBackend::Postgres, query))
-            .await
-            .map_err(|e| {
-                VectorDbError::IndexCreationError(format!("{:?}: {:?}", index_name.clone(), e))
-            })?;
-        // If more than one row is affected, something funky has occurred
-        if exec_res.rows_affected() > 1 {
-            return Err(VectorDbError::IndexCreationError(format!(
-                "More than one table affected! {:?} {:?}",
-                index_name.clone(),
-                exec_res.rows_affected()
-            )));
-        };
-        let query = r#"SET enable_seqscan = off;"#;
-        // The "enable seqscan = off" should not be set for the whole server. Will need to put this out
+
+        // Create index if dimensions is below 2000. It will be much slower for larger vectors, we should be explicit about this in the docs
+        // This is a limitation of pg_vector
+        if vector_dim <= 2000 {
```

Do this check at the top, so that if the dimensions are more than what we can handle we just throw an exception and the tables are never created. The user will then never be able to use the extractor unless they use a different storage. 

```
if vector_dim > 2000 { 

   // return Err 
}
```

---

i think with vector_dim > 2000 it still works, it will just be much slower because indexing is not on. do you want to throw an exception for postgres? it is a question of taste perhaps? can also throw an error as you like